src/delphi/train/permute.py

```diff
@@ -0,0 +1,36 @@
+# Number number generation utils without using any library
+
+
+class Permutation:
+    """
+    Permutation class
+    Atributes:
+    - state: int, seed for the random number generator
+
+    """
+
+    def __init__(self, seed):
+        self.state = seed
+
+    def set_seed(self, seed):
+        """
+        Set the seed
+        """
+        self.state = seed
+
+    def permute(self, list):
+        """
+        Permute a list in place using Fisher-Yates shuffle
+        """
+        for i in range(len(list)):
+            j = int(self.random_number_generator() * (i + 1))
```

Comment:
if random_number_generator() would return int then you just need to % i (or i+1?)

Reply:
I changed this to random uniform and created a generate_number, which generates a number from 0 to max (which was required for the correct shuffling algorithm)